src/codegen/c.zig

```diff
@@ -798,6 +795,131 @@ fn genStore(o: *Object, inst: *Inst.BinOp) !CValue {
+    if (!is_signed and inst.base.ty.tag() != .int_unsigned) {
+        return try genBinOp(o, inst, switch (op) {
+            .add => " + ",
+            .sub => " - ",
+            .mul => " * ",
+        });
+    }
+
+    var min_buf: [80]u8 = undefined;
+    const min = if (!is_signed)
+        "0"
+    else switch (inst.base.ty.tag()) {
+        .c_short => "SHRT_MIN",
+        .c_int => "INT_MIN",
+        .c_long => "LONG_MIN",
+        .c_longlong => "LLONG_MIN",
+        .isize => "INTPTR_MIN",
+        else => blk: {
+            // should be able to use undefined here since all the target specifics are handled
+            const bits = inst.base.ty.intInfo(@as(std.Target, undefined)).bits;
+            assert(bits <= 64); // TODO: large integers
+            const val = -1 * std.math.pow(i64, 2, @intCast(i64, bits - 1));
+            break :blk std.fmt.bufPrint(&min_buf, "{}", .{val}) catch |e|
+            // doesn't fit in some upwards error set, but should never happen
+                return if (e == error.NoSpaceLeft) unreachable else e;
+        },
+    };
+
+    var max_buf: [80]u8 = undefined;
+    const max = switch (inst.base.ty.tag()) {
+        .c_short => "SHRT_MAX",
+        .c_ushort => "USHRT_MAX",
+        .c_int => "INT_MAX",
+        .c_uint => "UINT_MAX",
+        .c_long => "LONG_MAX",
+        .c_ulong => "ULONG_MAX",
+        .c_longlong => "LLONG_MAX",
+        .c_ulonglong => "ULLONG_MAX",
+        .isize => "INTPTR_MAX",
+        .usize => "UINTPTR_MAX",
+        else => blk: {
+            // should be able to use undefined here since all the target specifics are handled
+            const bits = inst.base.ty.intInfo(@as(std.Target, undefined)).bits;
+            assert(bits <= 64); // TODO: large integers
+            const val = std.math.pow(u64, 2, if (is_signed) (bits - 1) else bits) - 1;
+            break :blk std.fmt.bufPrint(&max_buf, "{}", .{val}) catch |e|
+            // doesn't fit in some upwards error set, but should never happen
+                return if (e == error.NoSpaceLeft) unreachable else e;
+        },
+    };
+
+    const lhs = try o.resolveInst(inst.lhs);
+    const rhs = try o.resolveInst(inst.rhs);
+    const w = o.writer();
+
+    const ret = try o.allocLocal(inst.base.ty, .Mut);
+    try w.writeAll(" = zig_");
+    try w.writeAll(switch (op) {
+        .add => "addw_",
+        .sub => "subw_",
+        .mul => return o.dg.fail(.{ .node_offset = 0 }, "TODO: C backend: implement wrapping multiplication operator", .{}),
+    });
+
+    switch (inst.base.ty.tag()) {
+        .u8 => try w.writeAll("u8"),
+        .i8 => try w.writeAll("i8"),
+        .u16 => try w.writeAll("u16"),
+        .i16 => try w.writeAll("i16"),
+        .u32 => try w.writeAll("u32"),
+        .i32 => try w.writeAll("i32"),
+        .u64 => try w.writeAll("u64"),
+        .i64 => try w.writeAll("i64"),
+        .isize => try w.writeAll("isize"),
```

Comment:
is usize intentionally missing here?

Reply:
yes, it gets handled on line 811, since it's essentially a uintptr we just lean on C's existing wrapping for now